src/test/nft.t.sol

```diff
@@ -19,113 +19,203 @@ pragma experimental ABIEncoderV2;
 import "ds-test/test.sol";
 import "../nft.sol";
 
-contract User {
-    function doMint(address registry, address usr) public {
-    }
-}
-
 contract AnchorMock {
     bytes32 documentRoot;
     uint32  blockNumber;
 
     function file(bytes32 documentRoot_, uint32 blockNumber_) public {
         documentRoot = documentRoot_;
-        blockNumber = blockNumber;
+        blockNumber = blockNumber_;
     }
 
-    function getAnchorById(uint id) public returns (uint, bytes32, uint32) {
+    function getAnchorById(uint id) public view returns (uint, bytes32, uint32) {
         return (id, documentRoot, blockNumber);
     }
 }
 
+contract KeyManagerMock {
+    bytes32 key;
+    bytes32 value;
+    uint purpose;
+    uint[] mem;
+    bool validPurpose;
+    uint32 revoked;
+
+    function fileValidity(bool validity_) public {
```

Comment:
Can you follow this pattern https://github.com/centrifuge/tinlake/blob/master/src/maker.sol#L117 ?

Reply:
you can actually just use function overloading.
```suggestion
    function file(bool validity_) public {
```